cmd/earthly/root_cmds.go

```diff
@@ -173,6 +173,16 @@ func (app *earthlyApp) rootCmds() []*cli.Command {
 					Usage:       "Reset cache entirely by wiping cache dir",
 					Destination: &app.pruneReset,
 				},
+				&cli.StringFlag{
```

Comment:
I would rather create a flag value type for this size value:
```go
type byteSizeValue uint64

func (byteSizeValue) Create(val uint64, p *uint64, _ cli.NoConfig) cli.Value {
	*p = val
	return (*byteSizeValue)(p)
}

func (byteSizeValue) ToString(b uint64) string {
	return humanize.Bytes(b)
}

func (s *byteSizeValue) Set(s string) error {
	v, err := humanize.ParseBytes(s)
	if err != nil {
		return err
	}
	*s = byteSizeValue(v)
	return nil
}

func (s *byteSizeValue) Get() any { return uint64(*s) }

func (s *byteSizeValue) String() string { return humanize.Bytes(*s) }
```
And then to use it:
```go
&cli.FlagBase[uint64, cli.NoConfig, byteSizeValue]{
    Name: "size",
    Usage: "Prune cache to specified size, starting from oldest",
    Destination: &app.pruneTargetSize, // NOTE: pruneTargetSize would change to a uint64
}
```
This allows the `cli` library to know when the flag failed to parse, which _ideally_ means that it will be more portable and forward compatible and all that.

Reply:
Turns out you don't have to go that hard. They have a `GenericFlag` that works well enough with less code on our end.